sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -1500,16 +1511,65 @@ public void setTransactionProfiler(final @Nullable ITransactionProfiler transact
    * @return if profiling is enabled for transactions.
    */
   public boolean isProfilingEnabled() {
-    return profilingEnabled;
+    return (getProfilesSampleRate() != null && getProfilesSampleRate() > 0)
```

Comment:
We don't check `> 0` for `isTracingEnabled` at least.
Do we need it here?

Reply:
well, if the sample rate is 0 and there's no sampler set, we'd still initialize the profile, even if in reality it will never be used

Reply:
Yeah for profiling maybe it's better to keep with this check.
For transactions, we actually need it, because 0 is enabled and it propagates transactions that come from the backend and inherits the sampling decision from the parent.
If sampling should be the case, then removing the > 0 check makes sense.

Reply:
the logic of profiles sampling mimics the transaction sampling, including the parent decision. But we don't get profiles from the backend. Should we consider the case for consistency? Initializing the profiler is quick anyway, and the option already says that `null` should be used to disable it

Reply:
That's a business decision, so I can't tell you what makes more sense for profiling.